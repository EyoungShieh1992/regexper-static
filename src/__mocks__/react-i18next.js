const reactI18next = jest.requireActual('react-i18next');
const i18n = require('i18n');

module.exports = {
  ...reactI18next,
  Trans: require('__mocks__/component-mock').buildMock(reactI18next.Trans),
  useTranslation: () => ({ i18n, t: i18n.mockT })
};
