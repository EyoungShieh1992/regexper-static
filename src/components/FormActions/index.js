import React, { useState, useCallback, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useTranslation, Trans } from 'react-i18next';

import DownloadIcon from 'react-feather/dist/icons/download';
import LinkIcon from 'react-feather/dist/icons/link';

import style from './style.module.css';

import { createPngLink, createSvgLink } from './links';

const downloadLink = (link, t) => {
  const { url, filename, type, label } = link;
  return <li>
    <a href={ url } download={ filename } type={ type }>
      <DownloadIcon />{ t(label) }
    </a>
  </li>;
};

const FormActions = ({
  permalinkUrl,
  imageDetails
}) => {
  const { t } = useTranslation();
  const [svgLink, setSvgLink] = useState(null);
  const [pngLink, setPngLink] = useState(null);

  const generateDownloadLinks = useCallback(async () => {
    const { svg, width, height } = imageDetails;

    setSvgLink(await createSvgLink({ svg }));
    setPngLink(await createPngLink({ svg, width, height }));
  }, [setSvgLink, setPngLink, imageDetails]);

  useEffect(() => {
    if (imageDetails && imageDetails.svg) {
      generateDownloadLinks();
    }
  }, [imageDetails]);

  return <ul className={ style.actions }>
    { pngLink && downloadLink(pngLink, t) }
    { svgLink && downloadLink(svgLink, t) }
    { permalinkUrl && <li>
      <a href={ permalinkUrl }><LinkIcon /><Trans>Permalink</Trans></a>
    </li> }
  </ul>;
};

FormActions.propTypes = {
  permalinkUrl: PropTypes.string,
  imageDetails: PropTypes.shape({
    svg: PropTypes.string,
    width: PropTypes.number,
    height: PropTypes.number
  })
};

export default FormActions;
