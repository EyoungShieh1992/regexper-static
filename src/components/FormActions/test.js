jest.mock('./links');
jest.mock('react-feather/dist/icons/download', () =>
  require('__mocks__/component-mock')(
    'react-feather/dist/icons/download'));
jest.mock('react-feather/dist/icons/link', () =>
  require('__mocks__/component-mock')(
    'react-feather/dist/icons/link'));

import React from 'react';
import { render } from 'react-testing-library';

import FormActions from 'components/FormActions';
import { createPngLink, createSvgLink } from './links';

createPngLink.mockResolvedValue({
  url: 'http://example.com/image.png',
  filename: 'image.png',
  type: 'image/png',
  label: 'Example PNG Link'
});
createSvgLink.mockResolvedValue({
  url: 'http://example.com/image.svg',
  filename: 'image.svg',
  type: 'image/svg+xml',
  label: 'Example SVG Link'
});

describe('FormActions', () => {
  test('rendering', () => {
    const { asFragment } = render(
      <FormActions/>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('rendering with a permalink', () => {
    const { asFragment } = render(
      <FormActions permalinkUrl="http://example.com" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
