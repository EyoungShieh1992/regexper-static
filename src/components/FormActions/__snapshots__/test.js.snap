// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormActions rendering 1`] = `
<DocumentFragment>
  <ul
    class="actions"
  />
</DocumentFragment>
`;

exports[`FormActions rendering with a permalink 1`] = `
<DocumentFragment>
  <ul
    class="actions"
  >
    <li>
      <a
        href="http://example.com"
      >
        <span
          data-component="Link"
          data-props="{}"
        />
        <span
          data-component="Trans"
          data-props="{}"
        >
          Permalink
        </span>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
